### PR TITLE
Set RUBYLIB path correctly on alpine

### PR DESCRIPTION
**Pull request by kenhys**
GEM PATHS is different on alpine and debian.

Closes: #277

### CODE REVIEW

**Comment by kenhys** [comment]
I'll regenerate

**Comment by kenhys** [comment]
With modified image, RUBYLIB points it.

```
% docker run -v $PWD:/fluentd/etc -it tmp -c /fluentd/etc/tmp.conf
fluentd -c /fluentd/etc/tmp.conf
2021-06-21 08:39:53 +0000 [info]: parsing config file is succeeded path="/fluentd/etc/tmp.conf"
2021-06-21 08:39:53 +0000 [info]: gem 'fluent-plugin-record-modifier' version '2.1.0'
2021-06-21 08:39:53 +0000 [info]: gem 'fluentd' version '1.13.0'
2021-06-21 08:39:53 +0000 [warn]: both of Plugin @id and path for <storage> are not specified. Using on-memory store.
2021-06-21 08:39:53 +0000 [warn]: both of Plugin @id and path for <storage> are not specified. Using on-memory store.
2021-06-21 08:39:53 +0000 [warn]: define <match fluent.**> to capture fluentd logs in top level is deprecated. Use <label @FLUENT_LOG> instead
2021-06-21 08:39:53 +0000 [info]: using configuration file: <ROOT>
  <source>
    @type sample
    tag "sample"
  </source>
  <filter>
    @type record_modifier
    remove_keys "_dummy_"
    <record>
      _dummy_ ${record['loaded'] = $LOADED_FEATURES.collect { |entry| entry if entry.include?('resolv.rb') }.compact }
    </record>
  </filter>
  <match sample>
    @type stdout
  </match>
</ROOT>
2021-06-21 08:39:53 +0000 [info]: starting fluentd-1.13.0 pid=7 ruby="2.7.3"
2021-06-21 08:39:53 +0000 [info]: spawn command to main:  cmdline=["/usr/bin/ruby", "-Eascii-8bit:ascii-8bit", "/usr/bin/fluentd", "-c", "/fluentd/etc/tmp.conf", "-p", "/fluentd/plugins", "--under-supervisor"]
2021-06-21 08:39:53 +0000 [info]: adding filter pattern="**" type="record_modifier"
2021-06-21 08:39:53 +0000 [info]: adding match pattern="sample" type="stdout"
2021-06-21 08:39:53 +0000 [info]: adding source type="sample"
2021-06-21 08:39:53 +0000 [warn]: #0 both of Plugin @id and path for <storage> are not specified. Using on-memory store.
2021-06-21 08:39:53 +0000 [warn]: #0 both of Plugin @id and path for <storage> are not specified. Using on-memory store.
2021-06-21 08:39:53 +0000 [warn]: #0 define <match fluent.**> to capture fluentd logs in top level is deprecated. Use <label @FLUENT_LOG> instead
2021-06-21 08:39:53 +0000 [info]: #0 starting fluentd worker pid=16 ppid=7 worker=0
2021-06-21 08:39:53 +0000 [info]: #0 fluentd worker is now running worker=0
2021-06-21 08:39:53 +0000 [warn]: #0 no patterns matched tag="fluent.info"
2021-06-21 08:39:53 +0000 [warn]: #0 no patterns matched tag="fluent.info"
2021-06-21 08:39:54.032649090 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
2021-06-21 08:39:55.033852062 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
2021-06-21 08:39:56.035250257 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
2021-06-21 08:39:57.036810879 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
2021-06-21 08:39:58.038373954 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
2021-06-21 08:39:59.039944934 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
^C2021-06-21 08:39:59 +0000 [info]: Received graceful stop
2021-06-21 08:40:00.041698057 +0000 sample: {"message":"sample","loaded":["/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib/resolv.rb"]}
2021-06-21 08:40:00 +0000 [info]: #0 fluentd worker is now stopping worker=0
2021-06-21 08:40:00 +0000 [info]: #0 shutting down fluentd worker worker=0
2021-06-21 08:40:00 +0000 [info]: #0 shutting down input plugin type=:sample plugin_id="object:71c"
2021-06-21 08:40:00 +0000 [info]: #0 shutting down filter plugin type=:record_modifier plugin_id="object:794"
2021-06-21 08:40:00 +0000 [info]: #0 shutting down output plugin type=:stdout plugin_id="object:708"
2021-06-21 08:40:00 +0000 [info]: Worker 0 finished with status 0
```

File: v1.13/alpine/Dockerfile [hunk 1]
```diff
@@ -46,7 +46,7 @@ ENV FLUENTD_CONF="fluent.conf"
 ENV LD_PRELOAD=""
 # NOTE: resolv v0.2.1 includes the fix for CPU spike issue due to DNS resolver.
 # Forcing to load specific version of resolv (instead of bundled by default) is needed for Ruby 2.6.7, 2.7.3 and 3.0.1.
-ENV RUBYLIB="/usr/local/bundle/gems/resolv-0.2.1/lib"
+ENV RUBYLIB="/usr/lib/ruby/gems/2.7.0/gems/resolv-0.2.1/lib"
```
Comment (ashie) [inline v1.13/alpine/Dockerfile:49]: I think for v.1.12 is also needed.

Comment (kenhys) [reply v1.13/alpine/Dockerfile:49]: Yes, it should be too.

**Comment by kenhys** [comment]
It should rearrange commits a bit.

**Comment by kenhys** [comment]
squashed.

**Comment by kenhys** [comment]
I'll check CI failure.

**Comment by kenhys** [comment]
Does latest buster updates cause it? :thinking:

**Comment by kenhys** [comment]
It can be reproduced with local ruby:2.6-slim-buster image

```
wget -O /usr/local/bin/tini.asc https://github.com/krallin/tini/releases/download/v0.18.0/tini-amd64.asc
wget -O /usr/local/bin/tini https://github.com/krallin/tini/releases/download/v0.18.0/tini-amd64
export GNUPGHOME="$(mktemp -d)"
gpg --batch --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 6380DC428747F6C393FEACA59A84159D7001A4E5
gpg: keybox '/tmp/tmp.7bMJniiFST/pubring.kbx' created
gpg: keyserver receive failed: No name
```

**Comment by kenhys** [comment]
some weird things happen... :question: 

image

**Comment by cosmo0920** [comment]
Oh, GDPR took some effects for keyserver...! :cry:

**Comment by kenhys** [comment]
I've send a feedback to https://github.com/krallin/tini/issues/184

**Comment by kenhys** [comment]
I've skipped signed binary steps for a while.

**Comment by kenhys** [comment]
> I've skipped signed binary steps for a while.

extracted as a separate PR.

**Comment by kenhys** [comment]
I've split PR as https://github.com/fluent/fluentd-docker-image/pull/279 to focus on CI failure

**Comment by kenhys** [comment]
rebased because #279 was merged.

**Review by cosmo0920 — APPROVED** [review]
LGTM

**Comment by kenhys** [comment]
I'll push images.

**Comment by kenhys** [comment]
Kicked the build process.

image

**Comment by kenhys** [comment]
Done!

**Pull request merged**